### PR TITLE
[UR] Bump UR version and enable dynamic linking with UMF

### PR DESCRIPTION
**Pull request by igchor**
Testing PR for: https://github.com/oneapi-src/unified-runtime/pull/1430

### CODE REVIEW

**Comment by jsji** [comment]
@igchor Does this mean that we will be requiring ANYONE who use UR to install libhwloc on Linux and hwloc-win64-build-2.10.0 on windows? Can we somehow detect these, and don't switch to dynamic linking if these libs are not installed?

**Comment by igchor** [comment]
> @igchor Does this mean that we will be requiring ANYONE who use UR to install libhwloc on Linux and hwloc-win64-build-2.10.0 on windows? Can we somehow detect these, and don't switch to dynamic linking if these libs are not installed?

Yes, that's correct (although the minimum required version is 2.3.0). The hwloc is required not because we link dynamically with UMF but because I've bumped UMF version we are using. This new UMF version has a dependency on hwloc which cannot be disabled. Will this be a problem for intel/llvm?

**Comment by jsji** [comment]
> This new UMF version has a dependency on hwloc which cannot be disabled. Will this be a problem for intel/llvm?

Thanks @igchor  for the clarification. Installing hwloc in CI/containers won't be a problem. But that also means sycl compilers and DPC++ will have to add a dependency to hwloc? If so, I think we have to update quite a bunch of documents to reflect that? Is that really what we want?

FYI. @stdale-intel .

**Comment by stdale-intel** [comment]
@jsji  you are correct on both ends.  @uditagarwal97 updated the Windows CI machines, and Linux containers.  Docs and other things are being updated to reflect this as well.

**Comment by igchor** [comment]
> But that also means sycl compilers and DPC++ will have to add a dependency to hwloc?

Yes, and there's already been a discussion about how to handle this for DPC++.

> If so, I think we have to update quite a bunch of documents to reflect that? Is that really what we want?

@jsji do you want me to update any docs in intel/llvm in this PR? if yes, could you let me know which ones should I update?

**Comment by jsji** [comment]
> > But that also means sycl compilers and DPC++ will have to add a dependency to hwloc?
> 
> Yes, and there's already been a discussion about how to handle this for DPC++.
> 
> > If so, I think we have to update quite a bunch of documents to reflect that? Is that really what we want?
> 
> @jsji do you want me to update any docs in intel/llvm in this PR? if yes, could you let me know which ones should I update?

Thanks @igchor . I think @stdale-intel mentioned that "Docs and other things are being updated to reflect this as well." , not sure who is doing that, but I guess someone is already taking care of that.

**Comment by stdale-intel** [comment]
@igchor  if you can take care of the intel/llvm doc PR that would be awesome!

I'd start with editing https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md. But i'll see if there is other docs as well.

**Comment by igchor** [comment]
> @igchor if you can take care of the intel/llvm doc PR that would be awesome!
> 
> I'd start with editing https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md. But i'll see if there is other docs as well.

Done. Please let me know if I should update anything else.

**Comment by igchor** [comment]
@intel/llvm-gatekeepers, this PR is almost ready (I expect the required change to UR to be merged soon). Could you please let me know if everything looks good? Since this PR adds a new dependency on hwloc for sycl I would like to make sure this is fine and will not cause any issues (this was already reviewed by @stdale-intel a while ago).

**Comment by aelovikov-intel** [comment]
Awaiting @stdale-intel 's confirmation that downstream infrastructure has been updated to have `hwloc` installed on the systems before approving.

**Review by stdale-intel — APPROVED** [review]
intel/llvm and downstream infra has been updated to have required dependancies (hwloc)

**Review by pbalcer — APPROVED** [review]

**Comment by martygrant** [comment]
@pbalcer this is waiting on approvals from intel/dpcpp-devops-reviewers, intel/dpcpp-doc-reviewers, and/or intel/llvm-reviewers-runtime

**Comment by sarnex** [comment]
Don't we still need `dpcpp-doc-reviewers`?

**Comment by kswiecicki** [comment]
> Don't we still need `dpcpp-doc-reviewers`?

I'm not sure, @martygrant said `and/or` , we can wait for `intel/dpcpp-doc-reviewers` if it's necessary.

**Comment by sarnex** [comment]
The CI seems to be saying `and`, so @intel/dpcpp-doc-reviewers can someone please review this? Just a `apt-get` command change.

File: sycl/doc/GetStartedGuide.md
Comment (steffenlarsen) [inline sycl/doc/GetStartedGuide.md:46]:
```suggestion
* `hwloc` version 2.3 or later (Linux only)
```

**Review by steffenlarsen — APPROVED** [review]
I think we should make it clear that the `hwloc` prereq is a Linux-only requirement. Other than that, I think this is fine.

File: sycl/doc/GetStartedGuide.md
Comment (uditagarwal97) [reply sycl/doc/GetStartedGuide.md]: @igchor is `hwloc` a linux-only requirement? I recall installing `hwloc` in Windows CI machines...

Comment (igchor) [reply sycl/doc/GetStartedGuide.md]: We initially required hwloc on Windows but it turned out we couldn't support dynamic linking due to dependentloadflag flag being used. Since then we changed to linking statically with hwloc on windows (and so, user doesn't have to install anything now).

**Comment by igchor** [comment]
@intel/llvm-gatekeepers this is ready for merge, I belive we have all the required approvals now

**Comment by AlexeySachkov** [comment]
@igchor, this PR broke our post-commit, could you please take a look? https://github.com/intel/llvm/actions/runs/10106492701

```
FAILED: _deps/unified-memory-framework-build/src/CMakeFiles/umf.dir/provider/provider_os_memory.c.o 
ccache /opt/sycl/bin/clang -DUMF_SHARED_LIBRARY -DUMF_VERSION=0.9.0-rc1.git18.g1c34aae -DUR_ENABLE_SANITIZER -DUR_VALIDATION_LAYER_SUPPORTED_VERSION=\"0\" -DUR_VERSION=\"0\" -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Dumf_EXPORTS -I/__w/llvm/llvm/build/_deps/unified-memory-framework-build/src -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src -I/__w/llvm/llvm/build/include -I/__w/llvm/llvm/src/llvm/include -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/include -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/utils -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/base_alloc -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/critnib -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/provider -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memspaces -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memtargets -fPIC -fno-semantic-interposition -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -Wno-error -O3 -DNDEBUG -fPIC -fPIC -Wall -Wextra -Werror -Wpedantic -Wempty-body -Wunused-parameter -Wformat -Wformat-security -Wcast-qual -Wunused-result -MD -MT _deps/unified-memory-framework-build/src/CMakeFiles/umf.dir/provider/provider_os_memory.c.o -MF _deps/unified-memory-framework-build/src/CMakeFiles/umf.dir/provider/provider_os_memory.c.o.d -o _deps/unified-memory-framework-build/src/CMakeFiles/umf.dir/provider/provider_os_memory.c.o -c /__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/provider/provider_os_memory.c
/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/provider/provider_os_memory.c:182:5: error: default label in switch which covers all enumeration values [-Werror,-Wcovered-switch-default]
  182 |     default:
      |     ^
1 error generated.
[2193/8943] Building C object _deps/unified-memory-framework-build/src/CMakeFiles/umf.dir/memtargets/memtarget_numa.c.o
FAILED: _deps/unified-memory-framework-build/src/CMakeFiles/umf.dir/memtargets/memtarget_numa.c.o 
ccache /opt/sycl/bin/clang -DUMF_SHARED_LIBRARY -DUMF_VERSION=0.9.0-rc1.git18.g1c34aae -DUR_ENABLE_SANITIZER -DUR_VALIDATION_LAYER_SUPPORTED_VERSION=\"0\" -DUR_VERSION=\"0\" -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Dumf_EXPORTS -I/__w/llvm/llvm/build/_deps/unified-memory-framework-build/src -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src -I/__w/llvm/llvm/build/include -I/__w/llvm/llvm/src/llvm/include -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/include -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/utils -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/base_alloc -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/critnib -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/provider -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memspaces -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memtargets -fPIC -fno-semantic-interposition -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -Wno-error -O3 -DNDEBUG -fPIC -fPIC -Wall -Wextra -Werror -Wpedantic -Wempty-body -Wunused-parameter -Wformat -Wformat-security -Wcast-qual -Wunused-result -MD -MT _deps/unified-memory-framework-build/src/CMakeFiles/umf.dir/memtargets/memtarget_numa.c.o -MF _deps/unified-memory-framework-build/src/CMakeFiles/umf.dir/memtargets/memtarget_numa.c.o.d -o _deps/unified-memory-framework-build/src/CMakeFiles/umf.dir/memtargets/memtarget_numa.c.o -c /__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memtargets/memtarget_numa.c
/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memtargets/memtarget_numa.c:105:9: error: default label in switch which covers all enumeration values [-Werror,-Wcovered-switch-default]
  105 |         default:
      |         ^
/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memtargets/memtarget_numa.c:211:5: error: default label in switch which covers all enumeration values [-Werror,-Wcovered-switch-default]
  211 |     default:
      |     ^
/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memtargets/memtarget_numa.c:260:5: error: default label in switch which covers all enumeration values [-Werror,-Wcovered-switch-default]
  260 |     default:
      |     ^
3 errors generated.
[2194/8943] Building C object _deps/unified-memory-framework-build/src/pool/CMakeFiles/disjoint_pool.dir/__/__/__/unified-memory-framework-src/src/utils/utils_log.c.o
FAILED: _deps/unified-memory-framework-build/src/pool/CMakeFiles/disjoint_pool.dir/__/__/__/unified-memory-framework-src/src/utils/utils_log.c.o 
ccache /opt/sycl/bin/clang -DUMF_SHARED_LIBRARY -DUMF_VERSION=0.9.0-rc1.git18.g1c34aae -DUR_ENABLE_SANITIZER -DUR_VALIDATION_LAYER_SUPPORTED_VERSION=\"0\" -DUR_VERSION=\"0\" -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/__w/llvm/llvm/build/_deps/unified-memory-framework-build/src/pool -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/pool -I/__w/llvm/llvm/build/include -I/__w/llvm/llvm/src/llvm/include -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/include -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/utils -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/base_alloc -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/include/umf/pools -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/critnib -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/provider -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memspaces -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memtargets -fPIC -fno-semantic-interposition -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -Wno-error -O3 -DNDEBUG -fPIC -Wall -Wextra -Werror -Wpedantic -Wempty-body -Wunused-parameter -Wformat -Wformat-security -Wcast-qual -Wunused-result -MD -MT _deps/unified-memory-framework-build/src/pool/CMakeFiles/disjoint_pool.dir/__/__/__/unified-memory-framework-src/src/utils/utils_log.c.o -MF _deps/unified-memory-framework-build/src/pool/CMakeFiles/disjoint_pool.dir/__/__/__/unified-memory-framework-src/src/utils/utils_log.c.o.d -o _deps/unified-memory-framework-build/src/pool/CMakeFiles/disjoint_pool.dir/__/__/__/unified-memory-framework-src/src/utils/utils_log.c.o -c /__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/utils/utils_log.c
/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/utils/utils_log.c:62:5: error: default label in switch which covers all enumeration values [-Werror,-Wcovered-switch-default]
   62 |     default:
      |     ^
1 error generated.
```

**Comment by AlexeySachkov** [comment]
CI on macOS is also broken:

```
-- Checking for module 'hwloc>=2.3.0'
--   No package 'hwloc' found
-- Checking for module 'libhwloc' using find_library()
CMake Error at /Users/runner/work/llvm/llvm/build/_deps/unified-memory-framework-src/cmake/FindLIBHWLOC.cmake:71 (message):
  libhwloc NOT found (set CMAKE_PREFIX_PATH to point the location)
Call Stack (most recent call first):
  /Users/runner/work/llvm/llvm/build/_deps/unified-memory-framework-src/CMakeLists.txt:100 (find_package)
```

**Comment by steffenlarsen** [comment]
@pbalcer, would you be able to have a look at the above?

**Comment by pbalcer** [comment]
> @igchor, this PR broke our post-commit, could you please take a look? https://github.com/intel/llvm/actions/runs/10106492701
> 
> ```
> FAILED: _deps/unified-memory-framework-build/src/CMakeFiles/umf.dir/provider/provider_os_memory.c.o 
> ccache /opt/sycl/bin/clang -DUMF_SHARED_LIBRARY -DUMF_VERSION=0.9.0-rc1.git18.g1c34aae -DUR_ENABLE_SANITIZER -DUR_VALIDATION_LAYER_SUPPORTED_VERSION=\"0\" -DUR_VERSION=\"0\" -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Dumf_EXPORTS -I/__w/llvm/llvm/build/_deps/unified-memory-framework-build/src -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src -I/__w/llvm/llvm/build/include -I/__w/llvm/llvm/src/llvm/include -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/include -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/utils -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/base_alloc -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/critnib -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/provider -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memspaces -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memtargets -fPIC -fno-semantic-interposition -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -Wno-error -O3 -DNDEBUG -fPIC -fPIC -Wall -Wextra -Werror -Wpedantic -Wempty-body -Wunused-parameter -Wformat -Wformat-security -Wcast-qual -Wunused-result -MD -MT _deps/unified-memory-framework-build/src/CMakeFiles/umf.dir/provider/provider_os_memory.c.o -MF _deps/unified-memory-framework-build/src/CMakeFiles/umf.dir/provider/provider_os_memory.c.o.d -o _deps/unified-memory-framework-build/src/CMakeFiles/umf.dir/provider/provider_os_memory.c.o -c /__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/provider/provider_os_memory.c
> /__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/provider/provider_os_memory.c:182:5: error: default label in switch which covers all enumeration values [-Werror,-Wcovered-switch-default]
>   182 |     default:
>       |     ^
> 1 error generated.
> [2193/8943] Building C object _deps/unified-memory-framework-build/src/CMakeFiles/umf.dir/memtargets/memtarget_numa.c.o
> FAILED: _deps/unified-memory-framework-build/src/CMakeFiles/umf.dir/memtargets/memtarget_numa.c.o 
> ccache /opt/sycl/bin/clang -DUMF_SHARED_LIBRARY -DUMF_VERSION=0.9.0-rc1.git18.g1c34aae -DUR_ENABLE_SANITIZER -DUR_VALIDATION_LAYER_SUPPORTED_VERSION=\"0\" -DUR_VERSION=\"0\" -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Dumf_EXPORTS -I/__w/llvm/llvm/build/_deps/unified-memory-framework-build/src -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src -I/__w/llvm/llvm/build/include -I/__w/llvm/llvm/src/llvm/include -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/include -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/utils -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/base_alloc -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/critnib -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/provider -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memspaces -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memtargets -fPIC -fno-semantic-interposition -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -Wno-error -O3 -DNDEBUG -fPIC -fPIC -Wall -Wextra -Werror -Wpedantic -Wempty-body -Wunused-parameter -Wformat -Wformat-security -Wcast-qual -Wunused-result -MD -MT _deps/unified-memory-framework-build/src/CMakeFiles/umf.dir/memtargets/memtarget_numa.c.o -MF _deps/unified-memory-framework-build/src/CMakeFiles/umf.dir/memtargets/memtarget_numa.c.o.d -o _deps/unified-memory-framework-build/src/CMakeFiles/umf.dir/memtargets/memtarget_numa.c.o -c /__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memtargets/memtarget_numa.c
> /__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memtargets/memtarget_numa.c:105:9: error: default label in switch which covers all enumeration values [-Werror,-Wcovered-switch-default]
>   105 |         default:
>       |         ^
> /__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memtargets/memtarget_numa.c:211:5: error: default label in switch which covers all enumeration values [-Werror,-Wcovered-switch-default]
>   211 |     default:
>       |     ^
> /__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memtargets/memtarget_numa.c:260:5: error: default label in switch which covers all enumeration values [-Werror,-Wcovered-switch-default]
>   260 |     default:
>       |     ^
> 3 errors generated.
> [2194/8943] Building C object _deps/unified-memory-framework-build/src/pool/CMakeFiles/disjoint_pool.dir/__/__/__/unified-memory-framework-src/src/utils/utils_log.c.o
> FAILED: _deps/unified-memory-framework-build/src/pool/CMakeFiles/disjoint_pool.dir/__/__/__/unified-memory-framework-src/src/utils/utils_log.c.o 
> ccache /opt/sycl/bin/clang -DUMF_SHARED_LIBRARY -DUMF_VERSION=0.9.0-rc1.git18.g1c34aae -DUR_ENABLE_SANITIZER -DUR_VALIDATION_LAYER_SUPPORTED_VERSION=\"0\" -DUR_VERSION=\"0\" -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/__w/llvm/llvm/build/_deps/unified-memory-framework-build/src/pool -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/pool -I/__w/llvm/llvm/build/include -I/__w/llvm/llvm/src/llvm/include -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/include -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/utils -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/base_alloc -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/include/umf/pools -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/critnib -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/provider -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memspaces -I/__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/memtargets -fPIC -fno-semantic-interposition -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -Wno-error -O3 -DNDEBUG -fPIC -Wall -Wextra -Werror -Wpedantic -Wempty-body -Wunused-parameter -Wformat -Wformat-security -Wcast-qual -Wunused-result -MD -MT _deps/unified-memory-framework-build/src/pool/CMakeFiles/disjoint_pool.dir/__/__/__/unified-memory-framework-src/src/utils/utils_log.c.o -MF _deps/unified-memory-framework-build/src/pool/CMakeFiles/disjoint_pool.dir/__/__/__/unified-memory-framework-src/src/utils/utils_log.c.o.d -o _deps/unified-memory-framework-build/src/pool/CMakeFiles/disjoint_pool.dir/__/__/__/unified-memory-framework-src/src/utils/utils_log.c.o -c /__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/utils/utils_log.c
> /__w/llvm/llvm/build/_deps/unified-memory-framework-src/src/utils/utils_log.c:62:5: error: default label in switch which covers all enumeration values [-Werror,-Wcovered-switch-default]
>    62 |     default:
>       |     ^
> 1 error generated.
> ```

We are working on a fix. Will create a PR with a fix soon.

**Comment by pbalcer** [comment]
> CI on macOS is also broken:
> 
> ```
> -- Checking for module 'hwloc>=2.3.0'
> --   No package 'hwloc' found
> -- Checking for module 'libhwloc' using find_library()
> CMake Error at /Users/runner/work/llvm/llvm/build/_deps/unified-memory-framework-src/cmake/FindLIBHWLOC.cmake:71 (message):
>   libhwloc NOT found (set CMAKE_PREFIX_PATH to point the location)
> Call Stack (most recent call first):
>   /Users/runner/work/llvm/llvm/build/_deps/unified-memory-framework-src/CMakeLists.txt:100 (find_package)
> ```

This needs to be fixed by @intel/dpcpp-devops-reviewers by doing `brew install hwloc` on the mac runners.

**Comment by jsji** [comment]
> intel/llvm and downstream infra has been updated to have required dependancies (hwloc)


Note this is NOT true, it turns out that we haven't have hwloc installed in *ANY* of our internal build system, so I am going to revert this in sycl-web first to unblock pulldown.  This means this (and any new UR changes) will NOT be pulldown into downstream until we get the internal build system fixed , I have contacted our infrastructure team, hopefully they can help to install it ASAP.

**Pull request merged**
